### PR TITLE
R4R: Software upgrade test for v0.10

### PR DESCRIPTION
**Pull request by chengwenxi**
Refer to: https://github.com/irisnet/irishub/issues/993

**The difference between the v0.10-rc0 and the v0.10-rc1**:
*   Add `Text` type proposal 
*   Add `Votetime` be a part of `Vote`, use the machine system time which will cause the consensus failure

### CODE REVIEW

**Comment by haifengxi** [comment]
What logic has been changed in v1?  We need some change that can be used to confirm a successful upgrade.

**Comment by wukongcheng** [comment]
> What logic has been changed in v1? We need some change that can be used to confirm a successful upgrade.

Use the system time instead of BFT time to calculate the ending timestamp of the voting period. Therefore, the vote msg will cause the consensus failure. After that, the class3 upgrade drill is performed on this basis.

File: app/v1/gov/codec.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	cdc.RegisterConcrete(MsgDeposit{}, "irishub/gov/MsgDeposit", nil)
+	cdc.RegisterConcrete(MsgVote{}, "irishub/gov/MsgVote", nil)
+
+	cdc.RegisterInterface((*Proposal)(nil), nil)
+	cdc.RegisterConcrete(&TextProposal{}, "irishub/gov/TextProposal", nil)
+	cdc.RegisterConcrete(&ParameterProposal{}, "irishub/gov/ParameterProposal", nil)
+	cdc.RegisterConcrete(&SoftwareUpgradeProposal{}, "irishub/gov/SoftwareUpgradeProposal", nil)
+	cdc.RegisterConcrete(&SystemHaltProposal{}, "irishub/gov/SystemHaltProposal", nil)
+	cdc.RegisterConcrete(&TaxUsageProposal{}, "irishub/gov/TaxUsageProposal", nil)
+	cdc.RegisterConcrete(&Vote{}, "irishub/gov/Vote", nil)
```
Comment (wukongcheng) [inline app/v1/gov/codec.go]: RegisterConcrete() accept both value and pointer ? Which one should we use?

File: version/version.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 
 // Version - Iris Version
 const ProtocolVersion = 0
```
Comment (wukongcheng) [inline version/version.go]: The `ProtocolVersion` should be 1 now.

**Pull request merged**